Refactor module tests.foreman.ui.test_gpgkey

Ichimonji10 opened the pull request:
Use tools like the `setUp` function and for loops to reduce the repetitiveness
of tests. Use modern NailGun methods, doing things like replacing `create_json`
and `client.put` calls with `create` and `update` calls, respectively. These
changes make the module slightly longer, by about 10 lines. However, they also
drop the number of statements from 454 to 378, indicating much lower code
density.

Test results:

    (robottelo2)[ichimonji10@beech:robottelo]$ nosetests tests/foreman/ui/test_gpgkey.py
    .SSSS................S.S..S...S...S..S..S..

    ----------------------------------------------------------------------
    Ran 43 tests in 2144.077s

    OK (SKIP=11)

Ichimonji10 commented on tests/foreman/ui/test_gpgkey.py:
```diff
@@ -32,12 +40,10 @@ class GPGKey(UITestCase):
+    def setUpClass(cls):
+        cls.key_content = read_data_file(VALID_GPG_KEY_FILE)
+        cls.key_path = get_data_file(VALID_GPG_KEY_FILE)
+        cls.organization = entities.Organization().create()
```
I cannot set `cls.org`, as that attribute is already set by `UITestCase.setUp`. See [here](https://github.com/SatelliteQE/robottelo/blob/master/robottelo/test.py#L222).

elyezer commented on tests/foreman/ui/test_gpgkey.py:
```diff
@@ -111,16 +123,18 @@ def test_negative_create_2(self, name):
-            self.assertTrue(self.gpgkey.wait_until_element
-                            (common_locators["alert.error"]))
+            make_gpgkey(session, **kwargs)
+            self.assertTrue(
```
Do you mind update this to self.assertIsNotNone?

Ichimonji10 replied:
Sure, I'll update this assertion. Can you describe which assertions should be changed? I'm still learning my way around the UI test code.

Ichimonji10 replied:
I'm testing this diff now:

```diff
diff --git a/tests/foreman/ui/test_gpgkey.py b/tests/foreman/ui/test_gpgkey.py
index f437383..0c632fb 100644
--- a/tests/foreman/ui/test_gpgkey.py
+++ b/tests/foreman/ui/test_gpgkey.py
@@ -132,7 +132,7 @@ class GPGKey(UITestCase):
             make_gpgkey(session, **kwargs)
             self.assertIsNotNone(self.gpgkey.search(name))
             make_gpgkey(session, **kwargs)
-            self.assertTrue(
+            self.assertIsNotNone(
                 self.gpgkey.wait_until_element(common_locators['alert.error'])
             )
 
@@ -168,7 +168,7 @@ class GPGKey(UITestCase):
                 org=self.organization.name,
                 upload_key=True,
             )
-            self.assertTrue(
+            self.assertIsNotNone(
                 self.gpgkey.wait_until_element(common_locators['alert.error'])
             )
             self.assertIsNone(self.gpgkey.search(name))
@@ -191,7 +191,7 @@ class GPGKey(UITestCase):
                 org=self.organization.name,
                 upload_key=True,
             )
-            self.assertTrue(
+            self.assertIsNotNone(
                 self.gpgkey.wait_until_element(common_locators['haserror'])
             )
             self.assertIsNone(self.gpgkey.search(name))
@@ -213,7 +213,7 @@ class GPGKey(UITestCase):
                 name=name,
                 org=self.organization.name,
             )
-            self.assertTrue(
+            self.assertIsNotNone(
                 self.gpgkey.wait_until_element(common_locators['alert.error'])
             )
             self.assertIsNone(self.gpgkey.search(name))
@@ -286,7 +286,7 @@ class GPGKey(UITestCase):
             )
             self.assertIsNotNone(self.gpgkey.search(name))
             self.gpgkey.update(name, new_name)
-            self.assertTrue(self.gpgkey.wait_until_element(
+            self.assertIsNotNone(self.gpgkey.wait_until_element(
                 common_locators['alert.success']
             ))
 
@@ -314,7 +314,7 @@ class GPGKey(UITestCase):
             )
             self.assertIsNotNone(self.gpgkey.search(name))
             self.gpgkey.update(name, new_key=new_key_path)
-            self.assertTrue(self.gpgkey.wait_until_element(
+            self.assertIsNotNone(self.gpgkey.wait_until_element(
                 common_locators['alert.success']
             ))
 
@@ -338,7 +338,7 @@ class GPGKey(UITestCase):
             )
             self.assertIsNotNone(self.gpgkey.search(name))
             self.gpgkey.update(name, new_name)
-            self.assertTrue(self.gpgkey.wait_until_element(
+            self.assertIsNotNone(self.gpgkey.wait_until_element(
                 common_locators['alert.success']
             ))
 
@@ -365,7 +365,7 @@ class GPGKey(UITestCase):
             )
             self.assertIsNotNone(self.gpgkey.search(name))
             self.gpgkey.update(name, new_key=new_key_path)
-            self.assertTrue(self.gpgkey.wait_until_element(
+            self.assertIsNotNone(self.gpgkey.wait_until_element(
                 common_locators['alert.success']
             ))
 
@@ -392,7 +392,7 @@ class GPGKey(UITestCase):
             )
             self.assertIsNotNone(self.gpgkey.search(name))
             self.gpgkey.update(name, new_name)
-            self.assertTrue(
+            self.assertIsNotNone(
                 self.gpgkey.wait_until_element(common_locators['alert.error'])
             )
             self.assertIsNone(self.gpgkey.search(new_name))
@@ -417,7 +417,7 @@ class GPGKey(UITestCase):
             )
             self.assertIsNotNone(self.gpgkey.search(name))
             self.gpgkey.update(name, new_name)
-            self.assertTrue(
+            self.assertIsNotNone(
                 self.gpgkey.wait_until_element(common_locators['alert.error'])
             )
             self.assertIsNone(self.gpgkey.search(new_name))
```

rplevka replied:
:+1:

elyezer replied:
@Ichimonji10 sorry for the late reply but seems that you get all the assertions I meant. The thing is wait_until_element will return None if the element is not found and not a boolean. So is better to ensure that None is not returned. In practice this will not change because the python's feature to consider None as False. But this is intended to give more meaning when reading the code.

Ichimonji10 replied:
@elyezer Ahh. Thanks for the explanation. I agree with this change. The assertions are now more precise.

elyezer commented:
ACK + not blocker comment.

Ichimonji10 commented:
I've rebased and addressed @elyezer's comments. Here's the test results for updated methods:

```
(robottelo2)[ichimonji10@beech:robottelo]$ for meth in test_negative_create_2 test_negative_create_4 test_negative_create_5 test_negative_create_6 test_negative_update_1 test_negative_update_2 test_positive_update_1 test_positive_update_2 test_positive_update_3 test_positive_update_4; do nosetests tests/foreman/ui/test_gpgkey.py -m $meth; done
.
----------------------------------------------------------------------
Ran 1 test in 109.138s

OK
.
----------------------------------------------------------------------
Ran 1 test in 77.556s

OK
.
----------------------------------------------------------------------
Ran 1 test in 70.783s

OK
.
----------------------------------------------------------------------
Ran 1 test in 94.691s

OK
.
----------------------------------------------------------------------
Ran 1 test in 67.476s

OK
.
----------------------------------------------------------------------
Ran 1 test in 96.433s

OK
.
----------------------------------------------------------------------
Ran 1 test in 44.025s

OK
.
----------------------------------------------------------------------
Ran 1 test in 49.404s

OK
.
----------------------------------------------------------------------
Ran 1 test in 61.651s

OK
.
----------------------------------------------------------------------
Ran 1 test in 61.049s

OK
```

rplevka commented:
looking good,
ACK

elyezer commented:
:+1:
